File: Kata/FizzBuzz.cs
using System.Collections.Generic;
using System.Data.SqlTypes;
using System.Linq;

namespace Kata
{
    public class FizzBuzz
    {
        public string Play()
        {
            var ss = new List<string>();

            foreach (var i in Enumerable.Range(1, 100))
            {
                ss.Add(Next(i));
            }
            return string.Join("\r\n", ss);
        }

        private string Next(int i)
        {
            var s = "";
            if (i % 3 == 0) s = "Fizz";
            if (i % 5 == 0) s += "Buzz";
            if (i % 3 != 0 && i % 5 != 0) s += i.ToString();
            return s;
        }
    }
}

File: java/src/main/java/it/bebox/kata/FizzBuzz.java
package it.bebox.kata;

import org.apache.commons.lang3.StringUtils;

import java.util.*;

public class FizzBuzz {

    private final LinkedHashMap<Integer, String> rules;

    public FizzBuzz() {
        rules = new LinkedHashMap<Integer, String>() {{
            put(15, "FizzBuzz");
            put(3, "Fizz");
            put(5, "Buzz");
        }};
    }

    public String play() {
        List<String> results = new ArrayList<String>();

        for (int i = 1; i <= 100; i++) {

            int j = i;
            Optional<Map.Entry<Integer, String>> first = rules.entrySet().stream().filter(item -> canBeDividedBy(j, item.getKey())).findFirst();
            if (first.isPresent())
                results.add(first.get().getValue());
            else results.add(Integer.toString(i));
        }

        return StringUtils.join(results.iterator(), "\n");
    }

    private boolean canBeDividedBy(Integer i, Integer divisor) {
        return restOfDivision(i, divisor) == 0;
    }

    private int restOfDivision(Integer i, Integer divisor) {
        return i % divisor;
    }
}
